### ISSUE TITLE
compilation of nightly fails with RangeGeneratorInt message

### ISSUE THREAD

**Issue by NittanyLion** [JuliaStats/Distributions.jl#697]
My PR https://github.com/JuliaStats/Distributions.jl/pull/695 compiles fine on v0.6 but on nightly it fails with 

**ERROR: LoadError: LoadError: LoadError: UndefVarError: RangeGeneratorInt not defined**

The issue appears to arise in Categorical.jl, to which I have not made changes.

What gives?

**Comment by ararslan** [comment]
It's not an issue with your PR, it's because some existing code is doing `import Base: RangeGeneratorInt`, which doesn't exist on 0.7. I'm not entirely sure what the 0.7 replacement is, so I don't yet have a fix in mind. @rfourquet, you're the Keeper of Randomness in Base, do you have any sagely guidance for us?

**Comment by rfourquet** [comment]
For compatibility with "BaseBenchmarks", I have kept `RangeGenerator` in julia master as an alias for `Sampler`, but indeed `RangeGeneratorInt` was renamed to `SamplerRangeInt`. I see two solutions: have a conditional code which choose the correct name depending on the version, but I don't recommand this, as this is not yet official API and could be again renamed, and moreover the `Random` API moves towards having different sampler types which depend on the RNG. So the sanest option IMHO is to add a type parameter to `AliasTable` (which currently has a `RangeGeneratorInt` field) which will be the type of the sampler chosen by `RangeGenerator` (in 0.6) or `Sampler` in 1.0. I will be happy to help, but I can't install Distributions.jl on my system, so it's hard to prepare a PR and test it myself (Rmats doesn't install, I think it's a BinDeps problems, I'm on nixos :stuck_out_tongue: )

**Comment by NittanyLion** [comment]
Thank you both.  I'll wait for Alex to introduce the changes then.

**Comment by NittanyLion** [comment]
Actually, I managed to jump this hurdle, probably suboptimally.  

Now, nightly fails with

ERROR: LoadError: Base.nworkers has been moved to the standard library package Distributed.

which apparently is in deprecated.jl

Do I just wait for someone else to resolve this problem now?

**Comment by juliohm** [comment]
Having the same error on a fresh Julia v0.7 installation. Is the master branch ready? I am trying to port the GeoStats.jl framework to Julia v0.7.

**Comment by ararslan** [comment]
Not yet

**Comment by marcpabst** [comment]
This seems to be still a problem. Are there any news now that 0.7 beta is out?

My build on nightly still [fails](https://travis-ci.org/marcpabst/ANOVA.jl/jobs/398869835).

**Comment by ararslan** [comment]
No, 0.7 support here is still a work in progress. This is an enormous package with a lot of moving parts.

**Comment by rfourquet** [comment]
Out of the blue, I just got the idea to try installing Distributions again on my system (which never succeeded so far), and it worked flawlessly <3
It seems this issue has been fixed now? (probably by #729)
